spotify_scrape.py

```diff
@@ -0,0 +1,137 @@
+"""
+Uses the Spotify API, plus several Python libraries - Spotipy, a wrapper for the Spotify API,
+and Markovify, a library for Markov chains - to pull track names of a genre from Spotify,
+then use Markov chains to create random song names of a certain genre.
+
+@author: Matt Brucker
+"""
+
+import argparse
+from spotipy import Spotify
+import doctest
+import track_text
+import os.path
+from numpy.random import choice
+import matplotlib
+# Uncomment this line and change it if the default display backend doesn't work with matplotlib:
+# matplotlib.use('GTK3Cairo')
+import matplotlib.pyplot as plt
+
+
+# Parse arguments
+parser = argparse.ArgumentParser(description='Enter API Credentials')
+parser.add_argument('--fixed', help='Generate fixed-length track', action='store_true')
+parser.add_argument('--graph', help='Show a histogram of track lengths', action='store_true')
+parser.add_argument('genre', help='The genre of song to generate.')
+args = parser.parse_args()
+
+
+def save_track_list(client, genre):
+    """
+    Builds a pickle file containing the song names of up to 10,000 songs of a certain genre.
+    Strips extra words after hyphens from track names, and also converts everything to lowercase.
+    """
+    tracks = get_all_tracks(client, genre)
+    # Stores the data to reduce the number of API calls
+    file_name = '{}.txt'.format('_'.join(genre.lower().split()))
+    with open(file_name, 'w') as in_file:
+        in_file.write('\n'.join(tracks))
+
```

Comment:
After you open a file object, it is a good practice to close the file using `.close()`.